clang-tools-extra/docs/clang-tidy/checks/bugprone/use-after-move.rst

```diff
@@ -196,9 +196,10 @@ Any occurrence of the moved variable that is not a reinitialization (see below)
 is considered to be a use.
 
 An exception to this are objects of type ``std::unique_ptr``,
-``std::shared_ptr`` and ``std::weak_ptr``, which have defined move behavior
-(objects of these classes are guaranteed to be empty after they have been moved
-from). Therefore, an object of these classes will only be considered to be used
+``std::shared_ptr``, ``std::weak_ptr``, ``std::optional``, and ``std::any``,
+which have defined move behavior (objects of these classes are guaranteed to be
+empty after they have been moved from).
```

Comment:
An `optional` may technically not be empty when it previously had a value, its element would simply be moved from:

> Note that a moved-from optional still contains a value.

https://en.cppreference.com/w/cpp/utility/optional/operator%3D

Maybe move the sentence out of the parens and add 
> That is, [...] , and in the case ``std::optional``, the contained object is moved from.

Reply:
The docs look fine now, but I found an issue with adding `std::optional` and `std::any` like this